python/cudf/cudf/core/algorithms.py

```diff
@@ -59,3 +62,59 @@ def factorize(values, sort=False, na_sentinel=-1, size_hint=None):
     values.name = name
 
     return labels, cats.values if return_cupy_array else Index(cats)
+
+
+def linear_interpolation(to_interp):
+    """
+    Interpolate over a float column. Implicitly assumes that values are
+    evenly spaced with respect to the x-axis, for example the data
+    [1.0, NaN, 3.0] will be interpolated assuming the NaN is half way
+    between the two valid values, yielding [1.0, 2.0, 3.0]
+    """
+
+    to_interp._index = RangeIndex(start=0, stop=len(to_interp), step=1)
+    return index_or_values_interpolation(to_interp)
+
+
+def index_or_values_interpolation(to_interp):
```

Comment:
Same as above re:naming.
```suggestion
def _index_or_values_interpolation(to_interp):
```